tests/countries/test_palau.py

```diff
@@ -0,0 +1,163 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/python-holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.constants import ARMED_FORCES, HALF_DAY, PUBLIC
+from holidays.countries.palau import Palau, PW, PLW
+from tests.common import CommonCountryTests
+
+
+class TestPalau(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(Palau, years=range(1981, 2040), years_non_observed=range(1981, 2040))
+
+    def test_country_aliases(self):
+        self.assertAliases(Palau, PW, PLW)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Palau(years=1980))
+
+    def test_special_holidays(self):
+        self.assertHoliday(
+            "2020-11-03",
+        )
+
+    def test_armed_forces_holiday(self):
+        self.assertHolidays(
+            Palau(categories=ARMED_FORCES, years=2020), ("2020-11-11", "Veterans Day")
+        )
+
+    def test_half_day_holiday(self):
+        self.assertHolidays(
+            Palau(categories=HALF_DAY, years=2019),
+            ("2019-09-30", "Preparation for the 25th Independence Day of the Republic of Palau"),
+        )
+
+    def test_new_years_day_observed(self):
+        self.assertNoNonObservedHoliday(
+            "1982-12-31",
+            "1993-12-31",
+            "1999-12-31",
+            "2004-12-31",
+            "2010-12-31",
+            "2021-12-31",
+        )
+
+    def test_memorial_day(self):
+        name = "Memorial Day"
+        dt = (
+            # Last Monday of May.
+            "2011-05-30",
+            "2012-05-28",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertNoHolidayName(name, 2010, 2014)
```

Comment:
Let's make the tests more strict.

```suggestion
        self.assertNoHolidayName(name, range(1981, 2011), range(2013, 2040))
```